### PR TITLE
Move attribute naming docs from specification

### PR DESCRIPTION
**Pull request by jack-berg**
Related to https://github.com/open-telemetry/opentelemetry-specification/issues/3746. 

Moves [opentelemetry-specification/specification/common/attribute-naming.md](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-naming.md) and [opentelemetry-specification/specification/common/attribute-requirement-level.md](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-requirement-level.md) to `semantic-conventions`.

I've attempted to bring the commit history for these files along, but could use an extra set of eyes from @jsuereth, who I believe brought he git history with when seeding this repository.

### CODE REVIEW

**Comment by mx-psi** [comment]
I believe this fixes #295 cc @joaopgrassi

File: docs/general/attribute-naming.md
Comment (joaopgrassi) [inline docs/general/attribute-naming.md:68]: We can reference now the markdown file firectly

File: docs/general/attribute-requirement-level.md
Comment (joaopgrassi) [inline docs/general/attribute-requirement-level.md:22]: Now we will need to use the full link to the spec as we do elsewhere in the semconv repo. (referring to the spec version)

**Review by joaopgrassi — COMMENTED** [review]
Thank you!

The links needs to be adapted now, to link to the local files in this repo and use the full link to the spec repo.

File: docs/general/attribute-requirement-level.md
Comment (trask) [inline docs/general/attribute-requirement-level.md:1]:
```suggestion
# Attribute Requirement Levels
```

File: docs/general/attribute-naming.md
Comment (trask) [inline docs/general/attribute-naming.md:86]:
```suggestion
  namespaces ([Semantic Conventions](../README.md))
```

File: docs/general/attribute-naming.md
Comment (trask) [inline docs/general/attribute-naming.md:110]:
```suggestion
existing [semantic conventions](../README.md).
```

File: docs/general/attribute-requirement-level.md
Comment (trask) [inline docs/general/attribute-requirement-level.md:38]:
```suggestion
For example, [Database semantic convention](../database/README.md) references `network.transport` attribute defined in [General attributes](./README.md) with `Conditionally Required` level on it.
```

File: docs/general/attribute-requirement-level.md
Comment (trask) [inline docs/general/attribute-requirement-level.md:55]:
```suggestion
For example, `server.address` is `Conditionally Required` by the [Database convention](../database/README.md) when available. When `network.peer.address` is available instead, instrumentation can do a DNS lookup, cache and populate `server.address`, but only if the user explicitly enables the instrumentation to do so, considering the performance issues that DNS lookups introduce.
```

**Review by mx-psi — APPROVED** [review]

File: docs/general/attribute-requirement-level.md
Comment (joaopgrassi) [inline docs/general/attribute-requirement-level.md:22]:
```suggestion
Attribute requirement levels apply to the [instrumentation library](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.26.0/specification/glossary.md#instrumentation-library)).
```

**Review by joaopgrassi — APPROVED** [review]

**Comment by jack-berg** [comment]
FYI @open-telemetry/specs-semconv-maintainers, I believe that when it comes time to merge this, we should preserve the commit history instead of squashing.

**Review by AlexanderWert — APPROVED** [review]

File: docs/general/attribute-requirement-level.md
Comment (jsuereth) [inline docs/general/attribute-requirement-level.md:1]: We'll need to update all our internal links to the SPECIFICATION to this local document before you can delete the specification side, FYI.

That can be in this PR or a follow up.

**Review by pyohannes — APPROVED** [review]

File: docs/general/attribute-requirement-level.md
Comment (jack-berg) [reply docs/general/attribute-requirement-level.md:1]: I've pushed a commit which updates all the references to the spec version of these documents I could find 👍

**Comment by arminru** [comment]
@chalin FYI - see also https://github.com/open-telemetry/opentelemetry-specification/pull/3758 🙂

**Review by arminru — APPROVED** [review]

**Review by chalin — APPROVED** [review]
LGTM - it doesn't break anything when published to the OTel website ✨ 

One nit, see inline comment.

File: docs/general/attribute-requirement-level.md
Comment (jack-berg) [reply docs/general/attribute-requirement-level.md:1]: Per convo with @chalin, I've formatted these files using:
```
npm run _check:format:any -- --write docs/general/attribute-*.md --prose-wrap=always
```

**Comment by jack-berg** [comment]
Reminder to whoever merges this:

> FYI @open-telemetry/specs-semconv-maintainers, I believe that when it comes time to merge this, we should preserve the commit history instead of squashing.

**Comment by AlexanderWert** [comment]
> Reminder to whoever merges this:
> 
> > FYI @open-telemetry/specs-semconv-maintainers, I believe that when it comes time to merge this, we should preserve the commit history instead of squashing.

@arminru @jsuereth `Rebase and merge` is not enabled for this repository:
image

Can we enable it temporarily? Or how can we achieve the above otherwise?

**Comment by chalin** [comment]
The commit history remains available in the PR, why impose that on the main branch? For projects that are likely to have a long life it is preferable, IMHO, to keep the main branch's history as clean and terse as possible. (My 2 cents :)

**Comment by jsuereth** [comment]
> The commit history remains available in the PR, why impose that on the main branch? For projects that are likely to have a long life it is preferable, IMHO, to keep the main branch's history as clean and terse as possible. (My 2 cents :)

@chalin The goal here is to preserve the previous history from the spec repo for the files that are moved into this repository.

@AlexanderWert I can temporarily enable for this PR.

**Comment by chalin** [comment]
> @chalin The goal here is to preserve the previous history from the spec repo for the files that are moved into this repository.

Oh, got it! 👍🏻

**Review by jsuereth — APPROVED** [review]

**Comment by jsuereth** [comment]
> This branch cannot be rebased due to conflicts

@jack-berg  do you mind manually rebasing this and `push -f` on your private branch?

**Comment by jack-berg** [comment]
>> This branch cannot be rebased due to conflicts

> @jack-berg do you mind manually rebasing this and push -f on your private branch?

Done 👍

**Pull request merged**
